### DISCUSSION TITLE
need help with model ripping

### DISCUSSION THREAD

**Discussion by Soltinator** [flyinghead/flycast#809] [General]
i need some 3D models from a game.
i was able to capture a frame using nsight. but the geometry was flattened.

any ideas on how to unflatten? or what would cause this? or is there a better way to rip a model out of any obscure old dreamcast game?

**Comment by flyinghead** [comment]
The geometry sent by games to the dreamcast GPU is already projected in screen space so there's no easy way to get back to the 3D models.

It would probably be easier to find the model files by exploring the disk.

**Reply by Soltinator** [reply]
what would the hard way be? is there a way to intercept any geomerty on cpu before projection?
im assuming not all the games, save data the same. if not, could extracting vector3s from ram that connect together be a option?
 
i looked into it and the game i want to extract from has no data mine tools.
so i have to do with my armature coding skills.

**Reply by flyinghead** [reply]
you can do a reverse projection on the geometry captured (using an inverse projection matrix) since you still have the depth.

Another possibility is to reverse-engineer the game to find where the 3d models are stored in memory.

**Comment by vkedwardli** [comment]
https://youtu.be/YeQoaiVR4Wo
